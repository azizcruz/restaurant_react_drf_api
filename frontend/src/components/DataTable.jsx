import React, { Component } from "react";
import { Table, Icon } from "antd";

class DataTable extends Component {
  state = {
    filteredInfo: null,
    sortedInfo: null
  };

  handleChange = (pagination, filters, sorter) => {
    this.setState({
      filteredInfo: filters,
      sortedInfo: sorter
    });
  };

  render() {
    let { sortedInfo } = this.state;
    sortedInfo = sortedInfo || {};
    const columns = [
      {
        title: "Name",
        dataIndex: "name",
        key: "name",
        // Sort by the ascii code of the first name charachter.
        sorter: (a, b) => a.name.charCodeAt(0) - b.name.charCodeAt(0),
        sortOrder: sortedInfo.columnKey === "name" && sortedInfo.order
      },
      {
        title: "Price",
        dataIndex: "price",
        key: "price",
        render: text => <span>${text}</span>,
        // Sort by the price number.
        sorter: (a, b) => a.price - b.price,
        sortOrder: sortedInfo.columnKey === "price" && sortedInfo.order
      }, 
      {
        title: "Availability",
        dataIndex: "is_available",
        key: "is_available",
        render: (text) => (text) ?  <Icon type="check" /> :  <Icon type="close" />,
        // Sort by the price number.
        sorter: (a, b) => a.is_available - b.is_available,
        sortOrder: sortedInfo.columnKey === "is_available" && sortedInfo.order
      }
    ];
    return (
      <div>
        <Table
          columns={columns}
          dataSource={this.props.data}
          onChange={this.handleChange}
          rowKey="id"
          pagination={false}
          loading={this.props.isLoading}
          bordered={true}
        />
      </div>
    );
  }
}

export default DataTable;
