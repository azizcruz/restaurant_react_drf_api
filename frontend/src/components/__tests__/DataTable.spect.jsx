import React from "react";
import { create } from "react-test-renderer";
import DataTable from "../DataTable";

describe("DataTable component", () => {
    test("it matches the snapshot", () => {
      const component = create(<DataTable />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });


  describe("Datatable component", () => {
    it("shows a list of users", async () => {
      const component = create(<DataTable />);
      const instance = component.getInstance();
      expect(instance.state.filteredInfo).toBe(null)
      expect(instance.state.sortedInfo).toBe(null)
    });
  });