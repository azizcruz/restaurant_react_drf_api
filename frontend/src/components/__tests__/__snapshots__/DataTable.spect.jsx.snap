// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLayout component it matches the snapshot 1`] = `
<div>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div>
        <div
          className="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <div
        className="ant-spin-container ant-spin-blur"
      >
        <div
          className="ant-table ant-table-default ant-table-bordered ant-table-empty ant-table-scroll-position-left"
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              onWheel={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Name
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Price
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Availability
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                />
              </table>
            </div>
            <div
              className="ant-table-placeholder"
            >
              <div
                className="ant-empty ant-empty-normal"
              >
                <div
                  className="ant-empty-image"
                >
                  <img
                    alt="No Data"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                  />
                </div>
                <p
                  className="ant-empty-description"
                >
                  No Data
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DataTable component it matches the snapshot 1`] = `
<div>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-default ant-table-bordered ant-table-empty ant-table-scroll-position-left"
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              onWheel={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Name
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Price
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      className="ant-table-column-has-actions ant-table-column-has-sorters"
                      onClick={[Function]}
                    >
                      <span
                        className="ant-table-header-column"
                      >
                        <div
                          className="ant-table-column-sorters"
                        >
                          <span
                            className="ant-table-column-title"
                          >
                            Availability
                          </span>
                          <span
                            className="ant-table-column-sorter"
                          >
                            <div
                              className="ant-table-column-sorter-inner ant-table-column-sorter-inner-full"
                              title="Sort"
                            >
                              <i
                                aria-label="icon: caret-up"
                                className="anticon anticon-caret-up ant-table-column-sorter-up off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-up"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                  />
                                </svg>
                              </i>
                              <i
                                aria-label="icon: caret-down"
                                className="anticon anticon-caret-down ant-table-column-sorter-down off"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="caret-down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="0 0 1024 1024"
                                  width="1em"
                                >
                                  <path
                                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                  />
                                </svg>
                              </i>
                            </div>
                          </span>
                        </div>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                />
              </table>
            </div>
            <div
              className="ant-table-placeholder"
            >
              <div
                className="ant-empty ant-empty-normal"
              >
                <div
                  className="ant-empty-image"
                >
                  <img
                    alt="No Data"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                  />
                </div>
                <p
                  className="ant-empty-description"
                >
                  No Data
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
